### DISCUSSION TITLE
using where condition decimal numbers are treated as string

### DISCUSSION THREAD

**Discussion by kunalfleethawks** [sensedeep/dynamodb-onetable#117] [Q&A] [answered]
When we are using filter expression using find operation on model with where condition on number where value is whole number,  We are getting results back from dynamodb. But, when we use value as decimal number it is sent as string to dynamodb and thus we are not getting results back. 

### Working --  query with dynamodb one-table logs

![image](https://user-images.githubusercontent.com/60139157/131804771-26acdf3a-24db-4401-a86c-7c24a8586f36.png)

### NOT working 
Look at highlighted one in red. Library is sending it as quoted string as opposed to working use case where it is not quoted string in above screen shot
![image](https://user-images.githubusercontent.com/60139157/131805504-c43a56a5-dfcc-4405-9c16-55bbf6bd2ccd.png)

### Our Model class 
It is defined as number

![image](https://user-images.githubusercontent.com/60139157/131805226-039679d3-a6ee-4f16-9a88-45f0917bcb45.png)

Please let me know if you need more details 

@mobsense

**Comment by mobsense** [comment]
Thank you for reporting. That is a bug. The issue is using a floating point vs number. The current code supports numbers and not floating point. This fix addresses that.

The fix is:

```
diff --git a/src/Expression.js b/src/Expression.js
index f2f3efe..5be20e2 100644
--- a/src/Expression.js
+++ b/src/Expression.js
@@ -220,7 +220,7 @@ export class Expression {
         //  Expand value references and make attribute values
         where = where.replace(/{(.*?)}/g, (match, value) => {
             let index
-            if (value.match(/^\d+$/)) {
+            if (value.match(/^[-+]?([0-9]+(\.[0-9]*)?|\.[0-9]+)$/)) {
                 index = this.addValue(+value)
```

**Reply by kunalfleethawks** [reply]
Thank you. Can you let us know when the fix will be available ?

**Comment by mobsense** [comment, marked as answer]
Tomorrow. 1.7.6.

**Reply by kunalfleethawks** [reply]
Thank you @mobsense
